src/main/kotlin/org/veupathdb/service/eda/compute/EDA.kt

```diff
@@ -142,8 +143,15 @@ object EDA {
     // Create job ID by hashing plugin name and compute config
     val jobID = JobIDs.of(plugin.urlSegment, payload)
 
-    // If the job already exists, just return it.
-    AsyncPlatform.getJob(jobID)?.let { return it.toJobResponse() }
+    // Lookup the job to see if it already exists
+    val existingJob = AsyncPlatform.getJob(jobID)
+
+    // If the job already exists
+    if (existingJob != null)
+      // And the status is not expired OR autostart is false
+      if (existingJob.status != JS.Expired || !autostart)
```

Comment:
What's the reason behind adding the job status != expired check? If i'm reading it right, this means that when we ask about any expired jobs we'll always either start it or get a no-such-job status. If the check here was just on the autostart flag, then it seems like i could either get the normal job response or restart the job, based on the autostart flag?

@dmfalke and i discussed this a bit in the web-eda [PR](https://github.com/VEuPathDB/web-eda/pull/1644), but i think we agreed on having the frontend still ask about the job and get the expired status back, and then have the user click a button to specifically rerun the expired job (@dmfalke correct me if i misspoke!)

Reply:
What I was trying to implement is:

* If the job exists and is _not_ expired, then we return it.
* If the job exists and _is_ expired, then we test `autostart` and:
    * If `autostart` is `false` then we return the expired job.
    * If `autostart` is `true` then we continue on to submit the job.

Reply:
@Foxcapades Can you add JS.Failed here and solve the "restart failed jobs" issue?  I thought that needed an additional feature in lib-async-compute to clear out an individual job's workspace but if rerunning with expired works, will that also work?  Or is the issue that the expired job's workspace has already been cleared by the platform, so the new job execution is free to populate an empty space?